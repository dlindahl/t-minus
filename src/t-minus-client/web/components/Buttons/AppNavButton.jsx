import { assign } from 'lodash'
import { Children, cloneElement, PropTypes } from 'react'
import withSeverity from '../../decorators/withSeverity'

const baseStyles = {
  icon: {
    verticalAlign: 'top'
  },
  label: {
    fontFamily: 'Roboto Mono, monospace',
    fontSize: '.8rem',
    lineHeight: 1.5,
    textTransform: 'uppercase'
  },
  root: {
    background: 'transparent',
    border: 0,
    cursor: 'pointer',
    fontSize: '2rem',
    height: 50,
    lineHeight: 1,
    minWidth: 50,
    padding: 0
  }
}

function iconize (node) {
  return cloneElement(node, { style: baseStyles.icon })
}

const AppNavButton = (props) => {
  let color
  if (props.disabled) {
    color = props.textQuaternaryColor
  } else if (props.active) {
    color = props.textSecondaryColor
  } else {
    color = props.textTertiaryColor
  }

  const labelStyles = assign(
    {},
    baseStyles.label,
    { color }
  )
  return (
    <button
      onClick={props.onClick}
      style={
        assign(
          {},
          baseStyles.root,
          { color },
          props.style
        )
      }
    >
      {Children.map(props.children, iconize)}<br/>
      <div style={labelStyles}>
        {props.label}
      </div>
    </button>
  )
}

AppNavButton.propTypes = {
  active: PropTypes.bool,
  children: PropTypes.node,
  disabled: PropTypes.bool,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  style: PropTypes.object,
  textQuaternaryColor: PropTypes.string,
  textSecondaryColor: PropTypes.string,
  textTertiaryColor: PropTypes.string
}
AppNavButton.defaultProps = {
  active: false,
  disabled: false
}

export default withSeverity(AppNavButton)
