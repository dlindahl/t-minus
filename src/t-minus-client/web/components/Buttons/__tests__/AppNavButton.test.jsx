import AppNavButton from '../AppNavButton'
import Beer from 'react-icons/lib/fa/beer'
import configureStore from 'redux-mock-store'
import emptyObj from 'empty/object'
import renderer from 'react-test-renderer'

const STATE = {
  display: emptyObj
}
const store = configureStore()

describe('AppNavButton', () => {
  it('renders the component', () => {
    const component = renderer.create(
      <AppNavButton label="LABEL" store={store(STATE)}>
        <Beer/>
      </AppNavButton>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an active component', () => {
    const component = renderer.create(
      <AppNavButton active label="LABEL" store={store(STATE)}>
        <Beer/>
      </AppNavButton>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an inactive component', () => {
    const component = renderer.create(
      <AppNavButton active={false} label="LABEL" store={store(STATE)}>
        <Beer/>
      </AppNavButton>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a disabled component', () => {
    const component = renderer.create(
      <AppNavButton disabled label="LABEL" store={store(STATE)}>
        <Beer/>
      </AppNavButton>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an enabled component', () => {
    const component = renderer.create(
      <AppNavButton disabled={false} label="LABEL" store={store(STATE)}>
        <Beer/>
      </AppNavButton>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
