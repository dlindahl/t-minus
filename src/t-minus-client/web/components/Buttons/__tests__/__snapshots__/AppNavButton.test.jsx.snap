exports[`AppNavButton renders a disabled component 1`] = `
<button
  onClick={undefined}
  style={
    Object {
      "background": "transparent",
      "border": 0,
      "color": "var(--Clouds-Dark)",
      "cursor": "pointer",
      "fontSize": "2rem",
      "height": 50,
      "lineHeight": 1,
      "minWidth": 50,
      "padding": 0,
    }
  }>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "top",
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m14.8 20v-8.6h-5.7v5.7q0 1.2 0.8 2.1t2 0.8h2.9z m22.8 10v4.3h-25.7v-4.3l2.9-4.3h-2.9q-3.5 0-6-2.5t-2.5-6.1v-7.1l-1.5-1.4 0.7-2.9h10.8l0.7-2.8h21.4l0.7 4.2-1.4 0.8v17.8z" />
    </g>
  </svg>
  <br />
  <div
    style={
      Object {
        "color": "var(--Clouds-Dark)",
        "fontFamily": "Roboto Mono, monospace",
        "fontSize": ".8rem",
        "lineHeight": 1.5,
        "textTransform": "uppercase",
      }
    }>
    LABEL
  </div>
</button>
`;

exports[`AppNavButton renders an active component 1`] = `
<button
  onClick={undefined}
  style={
    Object {
      "background": "transparent",
      "border": 0,
      "color": "var(--Concrete)",
      "cursor": "pointer",
      "fontSize": "2rem",
      "height": 50,
      "lineHeight": 1,
      "minWidth": 50,
      "padding": 0,
    }
  }>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "top",
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m14.8 20v-8.6h-5.7v5.7q0 1.2 0.8 2.1t2 0.8h2.9z m22.8 10v4.3h-25.7v-4.3l2.9-4.3h-2.9q-3.5 0-6-2.5t-2.5-6.1v-7.1l-1.5-1.4 0.7-2.9h10.8l0.7-2.8h21.4l0.7 4.2-1.4 0.8v17.8z" />
    </g>
  </svg>
  <br />
  <div
    style={
      Object {
        "color": "var(--Concrete)",
        "fontFamily": "Roboto Mono, monospace",
        "fontSize": ".8rem",
        "lineHeight": 1.5,
        "textTransform": "uppercase",
      }
    }>
    LABEL
  </div>
</button>
`;

exports[`AppNavButton renders an enabled component 1`] = `
<button
  onClick={undefined}
  style={
    Object {
      "background": "transparent",
      "border": 0,
      "color": "var(--Silver)",
      "cursor": "pointer",
      "fontSize": "2rem",
      "height": 50,
      "lineHeight": 1,
      "minWidth": 50,
      "padding": 0,
    }
  }>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "top",
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m14.8 20v-8.6h-5.7v5.7q0 1.2 0.8 2.1t2 0.8h2.9z m22.8 10v4.3h-25.7v-4.3l2.9-4.3h-2.9q-3.5 0-6-2.5t-2.5-6.1v-7.1l-1.5-1.4 0.7-2.9h10.8l0.7-2.8h21.4l0.7 4.2-1.4 0.8v17.8z" />
    </g>
  </svg>
  <br />
  <div
    style={
      Object {
        "color": "var(--Silver)",
        "fontFamily": "Roboto Mono, monospace",
        "fontSize": ".8rem",
        "lineHeight": 1.5,
        "textTransform": "uppercase",
      }
    }>
    LABEL
  </div>
</button>
`;

exports[`AppNavButton renders an inactive component 1`] = `
<button
  onClick={undefined}
  style={
    Object {
      "background": "transparent",
      "border": 0,
      "color": "var(--Silver)",
      "cursor": "pointer",
      "fontSize": "2rem",
      "height": 50,
      "lineHeight": 1,
      "minWidth": 50,
      "padding": 0,
    }
  }>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "top",
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m14.8 20v-8.6h-5.7v5.7q0 1.2 0.8 2.1t2 0.8h2.9z m22.8 10v4.3h-25.7v-4.3l2.9-4.3h-2.9q-3.5 0-6-2.5t-2.5-6.1v-7.1l-1.5-1.4 0.7-2.9h10.8l0.7-2.8h21.4l0.7 4.2-1.4 0.8v17.8z" />
    </g>
  </svg>
  <br />
  <div
    style={
      Object {
        "color": "var(--Silver)",
        "fontFamily": "Roboto Mono, monospace",
        "fontSize": ".8rem",
        "lineHeight": 1.5,
        "textTransform": "uppercase",
      }
    }>
    LABEL
  </div>
</button>
`;

exports[`AppNavButton renders the component 1`] = `
<button
  onClick={undefined}
  style={
    Object {
      "background": "transparent",
      "border": 0,
      "color": "var(--Silver)",
      "cursor": "pointer",
      "fontSize": "2rem",
      "height": 50,
      "lineHeight": 1,
      "minWidth": 50,
      "padding": 0,
    }
  }>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "top",
      }
    }
    viewBox="0 0 40 40"
    width="1em">
    <g>
      <path
        d="m14.8 20v-8.6h-5.7v5.7q0 1.2 0.8 2.1t2 0.8h2.9z m22.8 10v4.3h-25.7v-4.3l2.9-4.3h-2.9q-3.5 0-6-2.5t-2.5-6.1v-7.1l-1.5-1.4 0.7-2.9h10.8l0.7-2.8h21.4l0.7 4.2-1.4 0.8v17.8z" />
    </g>
  </svg>
  <br />
  <div
    style={
      Object {
        "color": "var(--Silver)",
        "fontFamily": "Roboto Mono, monospace",
        "fontSize": ".8rem",
        "lineHeight": 1.5,
        "textTransform": "uppercase",
      }
    }>
    LABEL
  </div>
</button>
`;
